### PR TITLE
Feature/asynccluster

### PR DESCRIPTION
**Pull request by zfengms**
makeCluster return immediately when wait = TRUE parameter is specified, and add getCluster to check cluster creation status.

### CODE REVIEW

File: R/cluster.R [hunk 1]
```diff
@@ -225,34 +225,40 @@ makeCluster <-
...
-                             "deleting")) {
-        cat(".")
-        Sys.sleep(10)
-      }
+        cat(sprintf(message,
+                    poolConfig$name),
+            fill = TRUE)
 
-      cat("\n")
+        while (areShallowEqual(rAzureBatch::getPool(poolConfig$name)$state,
```
Comment (paselem) [inline R/cluster.R]: Do we need areShallowEqual here? Does '==' not work?

File: samples/long_running_job/long_running_job.R [hunk 2]
```diff
@@ -11,7 +11,7 @@ install_github("azure/doazureparallel")
 library(doAzureParallel)
 
 credentialsFileName <- "credentials.json"
-clusterFileName <- "cluster.json"
+clusterFileName <- "test_cluster.json"
```
Comment (paselem) [inline samples/long_running_job/long_running_job.R]: revert?

File: tests/testthat/test-async-cluster.R [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  doAzureParallel::generateClusterConfig(clusterFileName)
+
+  # set your credentials
+  doAzureParallel::setCredentials(credentialsFileName)
+
+  clusterName <-
+    doAzureParallel::makeCluster(clusterSetting = clusterFileName, wait = FALSE)
+
+  while (is.null(getCluster(clusterName))) {
+    Sys.sleep(30)
```
Comment (paselem) [inline tests/testthat/test-async-cluster.R]: Is this long enough? Are you only checking that the cluster object was created in Batch, or that it is fully provisioned?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,9 @@
 # Change Log
 
+## [0.6.2] 2018-01-05
```
Comment (paselem) [inline CHANGELOG.md]: If we are changing the version, you will also need to update the DESCRIPTION file as well.

File: R/cluster.R [hunk 5]
```diff
@@ -311,20 +317,102 @@ makeCluster <-
...
+#' Gets the cluster from your Azure account.
+#'
+#' @param clusterName The cluster configuration that was created in \code{makeCluster}
+#'
+#' @examples
+#' \dontrun{
+#' cluster <- getCluster("myCluster")
+#' }
+#' @export
+getCluster <- function(clusterName) {
```
Comment (brnleehng) [inline R/cluster.R]: Do we also need a listCluster function?

File: R/utility.R [hunk 6]
```diff
@@ -127,6 +80,64 @@ waitForNodesToComplete <- function(poolId, timeout = 86400) {
...
+      0
+    )
+
+    currentNodeCount <-
+      currentNodeCount + nodeValue
+  }
+  return(list(currentNodeCount = currentNodeCount, nodesWithFailures = nodesWithFailures))
+}
+
+.showNodesFailure <- function(nodesWithFailures) {
```
Comment (brnleehng) [inline R/utility.R]: We should print a brief message of the start task failure so users don't have to find the logs

File: R/cluster.R (hunk 1 above)
Comment (zfengms) [reply R/cluster.R]: This is existing code that I didn't modify, but I tested that '==' works fine, replaced it with '=='

File: CHANGELOG.md (hunk 4 above)
Comment (zfengms) [reply CHANGELOG.md]: The version is already 0.6.2 in DESCRIPTION file

File: samples/long_running_job/long_running_job.R (hunk 2 above)
Comment (zfengms) [reply samples/long_running_job/long_running_job.R]: reverted. updated all occurrence of test_cluster.json to cluster.json

File: tests/testthat/test-async-cluster.R (hunk 3 above)
Comment (zfengms) [reply tests/testthat/test-async-cluster.R]: updated getCluster to not check if the cluster is fully provisioned.

File: tests/testthat/test-async-cluster.R [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+    doAzureParallel::makeCluster(clusterSetting = clusterFileName, wait = FALSE)
+
+  cluster <- getCluster(cluster$poolId)
+  getClusterList()
+  filter <- filter <- list()
+  filter$state <- c("active", "deleting")
+  getClusterList(filter)
+  doAzureParallel::registerDoAzureParallel(cluster)
+
+  '%dopar%' <- foreach::'%dopar%'
```
Comment (brnleehng) [inline tests/testthat/test-async-cluster.R]: Remove? This should test only test getClusterList is working properly

**Pull request merged**